### PR TITLE
Add leaderboard submission scripts

### PR DESCRIPTION
**Pull request by melo-gonzo**
Adds two files to the `./scripts` folder which help in generating submissions to the evalai leaderboard in the correct format.
Fixes a bug in how `chunk_ids` were calculated.
Adds a case in dataset loading for differently formatted s2ef test data. 
Adds callbacks: leaderboard writer, gradient check, nan check (forward pass), and throughput.

### CODE REVIEW

File: ocpmodels/models/base.py [hunk 1]
```diff
@@ -860,6 +860,10 @@ def predict_step(
             fixed_mask = graph.ndata["fixed"] == 0
             # retrieve only forces corresponding to unfixed nodes
             predictions["forces"] = pred_force[fixed_mask]
+            natoms = tuple(batch.get('natoms').cpu().numpy().astype(int))
+            chunk_split = torch.split(graph.ndata["fixed"], natoms)
+            chunk_ids = [int((len(chunk)-sum(chunk)).cpu().numpy().astype(int)) for chunk in chunk_split]
```
Comment (laserkelvin) [inline ocpmodels/models/base.py:865]: Do you think you could split this up to multiple lines, just to make it easier to read?

```python
chunk_ids = []
for chunk in chunk_split:
    ids = (len(chunk) - sum(chunk)).cpu().numpy().astype(int)
    chunk_ids.append(int(ids))
```

Please check this actually produces what you intended!

File: scripts/generate_leaderboard_logs.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+from ocpmodels.lightning.cli import DATAMODULE_REGISTRY, MODEL_REGISTRY
+from ocpmodels.lightning.data_utils import IS2REDGLDataModule, is2re_devset
+from ocpmodels.models import GraphConvModel, IS2RELitModule
+from pytorch_lightning.loggers import CSVLogger
+from tqdm import tqdm
+from yaml import safe_load
+
+import os
+
+os.environ["TORCH_DISTRIBUTED_DEBUG"] = "DETAIL"  # set to DETAIL for runtime logging.
```
Comment (laserkelvin) [inline scripts/generate_leaderboard_logs.py:22]: Please remove this

**Review by laserkelvin — CHANGES_REQUESTED** [review]
Everything looks good; left two comments to be addressed and will be ready to merge after.

File: ocpmodels/models/base.py (hunk 1 above)
Comment (melo-gonzo) [reply ocpmodels/models/base.py]: Done!

File: scripts/generate_leaderboard_logs.py (hunk 2 above)
Comment (melo-gonzo) [reply scripts/generate_leaderboard_logs.py]: Done!

**Review by laserkelvin — APPROVED** [review]

**Pull request merged**